tensorflow/core/kernels/gpu_utils.h

```diff
@@ -91,15 +96,19 @@ class AutoTuneMap {
     return true;
   }
   void Insert(const Parameters& params, const Config& config) {
+    std::vector<std::unique_ptr<ConvolveExecutionPlan>> plans;
+    Insert(params, config, plans);
+  }
+  void Insert(const Parameters& params, const Config& config,
+              std::vector<std::unique_ptr<ConvolveExecutionPlan>>& plans) {
```

Comment:
As AutoTuneMap is a generic map from it's template type Parameter to template type Config, it shouldn't care about what is a `ConvolveExecutionPlan` - it's too specific. The user of this map should make the type Config contain all the information needed, including ConvolveExecutionPlan if necessary.

Can we make AlgorithmConfig contain `std::shared_ptr<const ConvolveExecutionPlan>`? Using shared pointer would still make AlgorithmConfig copyable.

Reply:
Done.